test/simulator/utils.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <initializer_list>
+#include <memory>
+#include <utility>
+
+#include "device.hpp"
+#include "simulator.hpp"
+
+namespace test {
+
+using two_way_link_t = std::pair<std::shared_ptr<sim::IRoutingDevice>,
+                                 std::shared_ptr<sim::IRoutingDevice>>;
+
+void add_links(sim::Simulator& sim,
+               std::initializer_list<two_way_link_t> links);
```

Comment:
Lets use name that speaks for itself better:
```suggestion
void add_two_way_links(sim::Simulator& sim,
               std::initializer_list<two_way_link_t> links);
```